operations/service_deploy_smoke_utils.go

```diff
@@ -68,7 +68,32 @@ func (tests smokeEndpointTestDefinitions) checkEndpoints() error {
 	return errors.Wrapf(catcher.Resolve(), "failed to get %d endpoints", catcher.Len())
 }
 
-func checkTaskByCommit(username, key, commit string) error {
+func getLatestGithubCommit() (string, error) {
+	client := util.GetHttpClient()
+	defer util.PutHttpClient(client)
+
+	resp, err := client.Get("https://api.github.com/repos/evergreen-ci/evergreen/git/refs/heads/master")
+	defer resp.Body.Close()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get latest commit from GitHub")
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", errors.Wrap(err, "error reading response body from GitHub")
+	}
+
+	latest := struct {
```

Comment:
I would recommend using github.Reference from go-github for this.

Reply:
Done.